src/openassetio-core/src/managerApi/ManagerInterface.cpp

```diff
@@ -14,11 +16,30 @@ namespace managerApi {
 
 ManagerInterface::ManagerInterface() = default;
 
+#define UNIMPLEMENTED_ERROR(capability)                                                           \
+  fmt::format(                                                                                    \
+      "The '{}' method has not been implemented by the manager. Check manager capability for {} " \
+      "by calling `manager.hasCapability`.",                                                      \
+      __func__, capability)
+
 InfoDictionary ManagerInterface::info() { return {}; }
 
-StrMap ManagerInterface::updateTerminology(StrMap terms,
+// NOLINTNEXTLINE(performance-unnecessary-value-param)
+void ManagerInterface::initialize(InfoDictionary managerSettings,
+                                  [[maybe_unused]] const HostSessionPtr& hostSession) {
+  if (!managerSettings.empty()) {
+    throw errors::InputValidationException{
+        "Settings provided but are not supported. The initialize method has not been implemented "
+        "by the manager."};
+  }
+}
+
+// NOLINTNEXTLINE(performance-unnecessary-value-param)
+StrMap ManagerInterface::updateTerminology([[maybe_unused]] StrMap terms,
                                            [[maybe_unused]] const HostSessionPtr& hostSession) {
-  return terms;
+  throw errors::NotImplementedException{
+      UNIMPLEMENTED_ERROR(ManagerInterface::kCapabilityNames[static_cast<int64_t>(
```

Comment:
[consider] Put the `static_cast` and name lookup inside the macro.

Reply:
Aye good shout https://github.com/OpenAssetIO/OpenAssetIO/pull/1141/commits/59c07d2ae8ea8fe9dc8a526ef807407eb106e47c